drivers/espi/espi_mchp_xec.c

```diff
@@ -899,10 +909,8 @@ static void espi_oob_down_isr(const struct device *dev)
 		/* Register is write-on-clear, ensure only 1 bit is affected */
 		ESPI_OOB_REGS->RX_STS = MCHP_ESPI_OOB_RX_STS_DONE;
 
-		/* Disable Rx interrupt */
-		ESPI_OOB_REGS->RX_IEN &= ~MCHP_ESPI_OOB_RX_IEN;
-
 		k_sem_give(&data->rx_lock);
```

Comment:
k_sem_give need to call only if other thread is waiting on k_sem_take.
if not, k_sem_give increments the count to 1 or greater for asynchronous events and when other thread perform sync OOB (send Tx first and waiting for Rx ),k_sem_take call returns immediately ( as the count is > 0) before response.

Reply:
Semaphores are not kernel primitives that are intended to indicate if something is waiting on them. 
But I understand the need to handle semaphore operations distinctly in master initiated OOB and slave initiated OOB transactions.